File: src/main/java/com/dude/dms/backend/data/rules/PlainTextRule.java
package com.dude.dms.backend.data.rules;

import com.dude.dms.backend.data.Tag;
import org.springframework.context.i18n.LocaleContextHolder;

import javax.persistence.Entity;
import javax.persistence.ManyToMany;
import javax.validation.constraints.NotBlank;
import java.util.Locale;
import java.util.Set;

@Entity
public class PlainTextRule extends Rule {

    @NotBlank
    private String text;

    private Boolean caseSensitive;

    @ManyToMany
    private Set<Tag> tags;

    public PlainTextRule() {
    }

    public PlainTextRule(@NotBlank String text, boolean caseSensitive, Set<Tag> tags) {
        super(true);
        this.text = text;
        this.caseSensitive = caseSensitive;
        this.tags = tags;
    }

    public String getText() {
        return text;
    }

    public void setText(String text) {
        this.text = text;
    }

    @Override
    public boolean validate(String line) {
        if (line != null && !line.isEmpty()) {
            if (caseSensitive) {
                return line.contains(text);
            } else {
                Locale locale = LocaleContextHolder.getLocale();
                return line.toLowerCase(locale).contains(text.toLowerCase(locale));
            }
        }
        return false;
    }

    @Override
    public Set<Tag> getTags() {
        return tags;
    }

    @Override
    public void setTags(Set<Tag> tags) {
        this.tags = tags;
    }

    @Override
    public String toString() {
        return "PlainTextRule{text='" + text + "'}";
    }

    public Boolean getCaseSensitive() {
        return caseSensitive;
    }

    public void setCaseSensitive(Boolean caseSensitive) {
        this.caseSensitive = caseSensitive;
    }
}

File: src/main/java/com/dude/dms/backend/data/rules/PlainTextRule.kt
package com.dude.dms.backend.data.rules

import com.dude.dms.backend.data.Tag
import org.springframework.context.i18n.LocaleContextHolder
import javax.persistence.Entity
import javax.persistence.ManyToMany

@Entity
class PlainTextRule(
        var text: String,
        var caseSensitive: Boolean,
        @ManyToMany override var tags: Set<Tag> = HashSet()
) : Rule() {

    override fun validate(line: String?): Boolean {
        return if (line != null && line.isNotEmpty()) {
            if (caseSensitive) text in line
            else {
                val locale = LocaleContextHolder.getLocale()
                text.toLowerCase(locale) in line.toLowerCase(locale)
            }
        } else false
    }

    override fun toString() = "PlainTextRule{text='$text'}"
}